js-pkg/packages/driftdb-react/src/index.tsx

```diff
@@ -108,11 +108,16 @@ export function useSharedState<T>(key: string, initialValue: T): [T, SetterFunct
   const db = useDatabase()
   const [state, setInnerState] = React.useState<T>(initialValue)
 
-  const stateListener = useRef<StateListener<T>>(null)
+  const stateListener = useRef<StateListener<SetStateAction<T>> | null>(null)
+
   if (stateListener.current === null) {
-    ;(stateListener as any).current = new StateListener(setInnerState, db, key)
+    stateListener.current = new StateListener(setInnerState, db, key)
   }
 
+  useEffect(() => {
+    stateListener.current!.subscribe()
```

Comment:
Another option here that might be a little simpler is to drop the `if (stateListener.current === null)` block above and just create the `StateListener` here in this `useEffect`. That way you don't need to separate out this `subscribe()` from the class's `constructor()`.

This would become:

```js
const stateListener = useRef<StateListener<SetStateAction<T>> | null>(null)
useEffect(() => {
  stateListener.current = new StateListener(setInnerState, db, key)
  return stateListener.current.destroy.bind(stateListener.current)
}, [])
```

Reply:
As discussed offline, we need the state listener for its `dispatch`